test-result-summary-client/src/PerfCompare/lib/ExtractRelevantJenkinsTestResults.js

```diff
@@ -0,0 +1,39 @@
+
+export default class ExtractRelevantJenkinsTestResults {
+    constructor(runJSON) {
+        this.runJSON = runJSON;
+        this.parsedVariants = [];
+    }
+
+    init(callback) {
+
+        let parsedVariantsCommon = {};
+        let curVariantObjectId = null;
+
+        if (Array.isArray(this.runJSON.testInfo[0].aggregateInfo) && this.runJSON.testInfo[0].aggregateInfo.length > 0) {
+            for (let aggregateInfo of this.runJSON.testInfo[0].aggregateInfo) {
+                if (aggregateInfo.benchmarkName && aggregateInfo.benchmarkVariant) {
+                    curVariantObjectId = aggregateInfo.benchmarkName + "!@#$%DELIMIT%$#@!" + aggregateInfo.benchmarkVariant;
+                } else {
+                    // invalid benchmark name and/or variant. Skipping it.
+                    continue;
+                }
+
+                // new variant
+                if (parsedVariantsCommon[curVariantObjectId] === undefined) {
+                    parsedVariantsCommon[curVariantObjectId] = {
+                        product: (aggregateInfo.benchmarkProduct === undefined) ? null : aggregateInfo.benchmarkProduct,
+                        benchmark: (aggregateInfo.benchmarkName === undefined) ? null : aggregateInfo.benchmarkName,
+                        variant: (aggregateInfo.benchmarkVariant === undefined) ? null : aggregateInfo.benchmarkVariant,
+                        machine: (this.runJSON.testInfo[0].machine === null) ? null : this.runJSON.testInfo[0].machine.split('.')[0],
```

Comment:
The machine is on which the builds are running. Also we would like to display it the perf compare GUI to let the users know directly the platforms or machines used. 
The machine value parsed in the database looks like this `kermit.locationlab.company.com`, and split it to keep `kermit` only to avoid info disclosure.

Reply:
@sophiaxu0424 We can disclose that info since TRSS could be deployed at different servers and it's fine to disclose that info to anyone who has access to that particular instance of TRSS. So let's show the full name along with domain name for the machine.